src/functional/iterator/embedded.py

```diff
@@ -370,8 +371,7 @@ def execute_shift(
     assert tag in offset_provider
     offset_implementation = offset_provider[tag]
     if isinstance(offset_implementation, Dimension):
-        assert offset_implementation.value in pos
-        new_pos = pos.copy()
+        new_pos = copy.copy(pos)
```

Comment:
Is there any difference between `copy.copy(dict`) and `dict.copy()`?

Reply:
Only with respect to typing. `Mapping` does not guarantee `mapping.copy` to exist.